vulture/utils.py

```diff
@@ -63,23 +65,32 @@ def get_decorator_name(decorator):
     return "@" + ".".join(reversed(parts))
 
 
-def get_modules(paths, toplevel=True):
+def get_modules(paths):
     """Take files from the command line even if they don't end with .py."""
     modules = []
     for path in paths:
-        path = os.path.abspath(path)
-        if toplevel and path.endswith(".pyc"):
-            sys.exit(f".pyc files are not supported: {path}")
-        if os.path.isfile(path) and (path.endswith(".py") or toplevel):
-            modules.append(path)
-        elif os.path.isdir(path):
-            subpaths = [
-                os.path.join(path, filename)
-                for filename in sorted(os.listdir(path))
-            ]
-            modules.extend(get_modules(subpaths, toplevel=False))
-        elif toplevel:
+        path = path.resolve()
+
+        if not path.exists():
             sys.exit(f"Error: {path} could not be found.")
+
+        if path.is_file():
+            top_paths = (_ for _ in [path])
+        else:
+            top_paths = path.glob("*")
```

Comment:
The new code seems overly complicated. Basically, it should do: loop over all given paths, add all *files* that end with .py, abort if any ends with .pyc and add all .py files under the given *directories*.

Reply:
The function did look long but I couldn't think of a simpler way.

What I tried to do there is as the way that you described.

Any ideas on making it simpler?

Reply:
I'll have a look when I find the time. Can you make sure that all tests pass?